### ISSUE TITLE
(sccz80) Support bitfields

### ISSUE THREAD

**Issue by suborb** [z88dk/z88dk#1133]
A long standing omission, it's the last one after #1132 is merged.

I've not actually seen any requests or complaints about them being missing over the past 20 or so years but let's see what can be done to add them.

Remaining:

- [x] Parsing bitfields
- [x] Parsing padding bitfields
- [x] Storing a bit field
- [x] Retrieving an unsigned bitfield
- [x] Retrieving a signed bitfield
- [x] Emulate zsdcc padding/sign rules
- [x] Initialisation
- [x] Range checking

**Comment by suborb** [comment]
There's some interesting behaviour in sdcc which I should emulate to avoid porting issues:

```
struct {
        signed int      a:5;
        int     b:4;
        int     c:5;
} x;
```

This has a size of 3 bytes (so a, b, c are held in different bytes). b and c are unsigned, a is signed.

In the branch at the moment, in sccz80 this is packed into two bytes and all values are unsigned.

**Comment by suborb** [comment]
Merged as #1143
